src/main/java/org/openpnp/gui/JobPlacementsPanel.java

```diff
@@ -39,9 +39,11 @@
 import org.openpnp.gui.support.IdentifiableListCellRenderer;
 import org.openpnp.gui.support.IdentifiableTableCellRenderer;
 import org.openpnp.gui.support.MessageBoxes;
+import org.openpnp.gui.support.PartCellValue;
 import org.openpnp.gui.support.PartsComboBoxModel;
 import org.openpnp.gui.tablemodel.PlacementsTableModel;
 import org.openpnp.gui.tablemodel.PlacementsTableModel.Status;
+import org.openpnp.machine.reference.ReferencePnpJobProcessor;
```

Comment:
UI classes like JobPlacementsPanel cannot access implementation classes in this way. You must go through the SPI interface http://openpnp.github.io/openpnp/develop/org/openpnp/spi/PnpJobProcessor.html. The reason for this is that users can change which job processor they use by changing the class in machine.xml. This would make the feature fail to work in any setup that did not use the default.

Instead, add getJobPlacements() to the PnpJobProcessor interface and then access it via machine.getPnpJobProcessor(). Then do the various bits of filtering in UI code rather than in the JobProcessor so that we don't have to expose UI specific methods in the SPI class.